src/lib/crypto/hash.cpp

```diff
@@ -222,7 +233,19 @@ pgp_hash_add_int(pgp_hash_t *hash, unsigned n, size_t length)
 int
 pgp_hash_add(pgp_hash_t *hash, const void *buf, size_t len)
 {
-    return botan_hash_update((botan_hash_t) hash->handle, (const uint8_t *) buf, len);
+    if (!hash->handle) {
+        return -1;
+    }
+
+    int error = 0;
```

Comment:
This variable seems unnecessary, I'd rather we drop it and `return -2` directly in the catch body.